### DISCUSSION TITLE
firewallcmd-ipset logging

### DISCUSSION THREAD

**Discussion by ghost** [fail2ban/fail2ban#3834] [Q&A]
Hello!

What is the proper way to enable logging with the firewallcmd-ipset action?
I would like to log any rejected packages via nftables.

I do not want to manually edit nftables whenever possible.

The chains from firewalld have the following line:
`log prefix "filter_IN_public_REJECT: "`
I would like to add such a line to the f2b stuff, so I can see in the dmesg when packages are rejected.

-- 
kind regards

**Comment by sebres** [comment]
Isn't this a matter of `firewallcmd-rich-rules` or rather a `firewallcmd-rich-logging` actions?
I'm unsure it is possible at all with stock `firewallcmd-ipset` action, just because it uses `firewall-cmd --add-rule` and `firewall-cmd --add-rich-rule`, but I can be wrong, since I never used it (why someone needs this weird mockup-proxy over a native subsystem as fail2ban action?).
Probably you have to create new action mixing both in single one.

But since firewalld simply wraps the rule to iptables/nftables backends, one could try to rewrite action `blocktype` parameter in order to get something like:
```
firewall-cmd --direct --add-rule ... -j LOG --log-prefix "filter_IN_public_REJECT: " --log-level 4
```
in this case it'd be:
```ini
banaction = firewallcmd-ipset[blocktype='LOG --log-prefix "filter_IN_public_REJECT: " --log-level 4']
```
but it'd then miss the default blocktype (and therefore doesn't have `-j REJECT --reject-with <rejecttype>`)... 

This would be possible too:
```ini
banaction = firewallcmd-ipset[blocktype='<known/blocktype> LOG --log-prefix "filter_IN_public_REJECT: " --log-level 4']
```
just it'd generate two `-j ...` parameters (one with REJECT another with LOG) for firewall-cmd, and I don't know how it'd behave in that case.

Maybe one could somehow add `--set-log-denied=on` parameter, but I don't know which prefix it'd use in case of reject.

Anyway if you know the proper syntax of firewall-cmd command for that, then provide them completely, so I could help to adjust firewallcmd-ipset or create a derivative action for it.
